lib/Conversion/AffineToPipeline/AffineToPipeline.cpp

```diff
@@ -52,17 +54,47 @@ struct AffineToPipeline : public AffineToPipelineBase<AffineToPipeline> {
   void runOnOperation() override;
 
 private:
+  ModuloProblem getModuloProblem(CyclicProblem &prob);
   LogicalResult
   lowerAffineStructures(MemoryDependenceAnalysis &dependenceAnalysis);
-  LogicalResult populateOperatorTypes(SmallVectorImpl<AffineForOp> &loopNest);
-  LogicalResult solveSchedulingProblem(SmallVectorImpl<AffineForOp> &loopNest);
-  LogicalResult createPipelinePipeline(SmallVectorImpl<AffineForOp> &loopNest);
+  LogicalResult populateOperatorTypes(SmallVectorImpl<AffineForOp> &loopNest,
+                                      ModuloProblem &problem);
```

Comment:
Maybe the `ModuloProblem` could be an instance variable instead of passing it around?

Also--and this can be separate--I think it would make sense to have the SchedulingAnalysis just create a `ModuloProblem` instead of a `CyclicProblem`.

Reply:
For context, that analysis only exists for this pass currently, so if moving the goalposts simplifies things, that seems fine.